py_zipkin/encoding/_encoders.py

```diff
@@ -95,20 +115,13 @@ def encode_span(self, span_builder):
             thrift_endpoint,
         )
 
-        # Add sa binary annotation
-        if span.sa_endpoint is not None:
-            thrift_sa_endpoint = thrift.create_endpoint(
-                span.sa_endpoint.port,
-                span.sa_endpoint.service_name,
-                span.sa_endpoint.ipv4,
-                span.sa_endpoint.ipv6,
+        # Add sa/ca binary annotations
+        if v2_span.remote_endpoint:
+            self.encode_remote_endpoint(
```

Comment:
maybe check empty objects or null keys aren't made here. better to send less data

Reply:
which empty objects or null keys are you referring to?

Reply:
That's kind of taken care in `py_zipkin/thrift/__init__.py`. However the thrift constructor seems to require all arguments, so it'll depend on what that does. Hopefully it'll just ignore all the `None` arguments.

This is also how it already works right now and given this encoding is deprecated I'm not too interested in optimizing it.